argh_derive/src/lib.rs

```diff
@@ -291,7 +291,18 @@ fn impl_from_args_struct(
                     } else {
                         __remaining_args
                     };
-                    #name = Some(<#ty as argh::FromArgs>::from_args(&__command, __remaining_args)?);
+                    if __dump_args {
+                        __args_dump.push(__subcommand.name.to_string().replace(" ", "\\ "));
```

Comment:
As mentioned below, I think we should keep the arguments as a vector to allow callers to escape arguments if necessary.